src/solver/optimisation/constraint_builder.h

```diff
@@ -0,0 +1,376 @@
+#pragma once
```

Comment:
We should rename the file accordingly with the convention used in other new files (CamelCase), since we seem to be adopting that convention.

Reply:
done